lib/internal/modules/esm/translators.js

```diff
@@ -501,19 +509,59 @@ translators.set('wasm', async function(url, source) {
     throw err;
   }
 
-  const imports =
-      ArrayPrototypeMap(WebAssembly.Module.imports(compiled),
-                        ({ module }) => module);
-  const exports =
-    ArrayPrototypeMap(WebAssembly.Module.exports(compiled),
-                      ({ name }) => name);
+  const importsList = new SafeSet();
+  const wasmGlobalImports = [];
+  for (const impt of WebAssembly.Module.imports(compiled)) {
+    if (impt.kind === 'global') {
+      ArrayPrototypePush(wasmGlobalImports, impt);
+    }
+    importsList.add(impt.module);
+  }
+
+  const exportsList = new SafeSet();
+  const wasmGlobalExports = new SafeSet();
+  for (const expt of WebAssembly.Module.exports(compiled)) {
+    if (expt.kind === 'global') {
+      wasmGlobalExports.add(expt.name);
+    }
+    exportsList.add(expt.name);
+  }
 
-  const createDynamicModule = require(
-    'internal/modules/esm/create_dynamic_module');
-  const { module } = createDynamicModule(imports, exports, url, (reflect) => {
+  const createDynamicModule = require('internal/modules/esm/create_dynamic_module');
+
+  const { module } = createDynamicModule([...importsList], [...exportsList], url, (reflect) => {
```

Comment:
Spread might not be safe even on Safe* objects. @aduh95 I think you know better.

Reply:
Marking this one as resolved - we use spread throughout the codebase. But if anyone has any concerns please do raise them.